gnome-initial-setup/pages/account/gis-account-page.c

```diff
@@ -161,12 +162,67 @@ switch_login_mode (GisAccountPage *page)
     set_mode (page, UM_LOCAL);
 }
 
+#define DEMO_MODE_ENTERED_METRIC "c75af67f-cf2f-433d-a060-a670087d93a1"
+
+static void
+enable_endlessm_metrics (gboolean demo_mode)
+{
+  GDBusProxy *metrics_proxy;
+  EmtrEventRecorder *recorder;
+  GVariantDict dict;
+  GError *error = NULL;
+
+  metrics_proxy = g_dbus_proxy_new_for_bus_sync (G_BUS_TYPE_SYSTEM,
+                                                 G_DBUS_PROXY_FLAGS_NONE,
+                                                 NULL,
+                                                 "com.endlessm.Metrics",
+                                                 "/com/endlessm/Metrics",
+                                                 "com.endlessm.Metrics.EventRecorderServer",
+                                                 NULL, &error);
+
+  if (error != NULL)
+    {
+      g_critical ("Unable to create a DBus proxy for the metrics daemon: %s", error->message);
+      g_clear_object (&metrics_proxy);
+      g_clear_error (&error);
+      return;
+    }
+
+  g_dbus_proxy_call_sync (metrics_proxy,
+                          "SetEnabled",
+                          g_variant_new ("(b)", TRUE),
+                          G_DBUS_CALL_FLAGS_NONE, -1,
+                          NULL, &error);
+  g_clear_object (&metrics_proxy);
+
+  if (error != NULL)
+    {
+      g_critical ("Unable to set the enabled state of metrics daemon: %s", error->message);
+      g_clear_error (&error);
+      return;
+    }
+
+  if (demo_mode)
+    {
+      recorder = emtr_event_recorder_get_default ();
+
+      g_variant_dict_init (&dict, NULL);
+
+      emtr_event_recorder_record_event (recorder,
+                                        DEMO_MODE_ENTERED_METRIC,
+                                        g_variant_dict_end (&dict));
```

Comment:
Could you not pass `NULL` or just an empty variant instead of creating this empty dict here?